src/libcollections/vec_deque.rs

```diff
@@ -1243,14 +1242,19 @@ impl<T> VecDeque<T> {
     /// Panics if `index` is greater than `VecDeque`'s length
     ///
     /// # Examples
+    ///
     /// ```
     /// use std::collections::VecDeque;
     ///
-    /// let mut buf = VecDeque::new();
-    /// buf.push_back(10);
-    /// buf.push_back(12);
-    /// buf.insert(1, 11);
-    /// assert_eq!(Some(&11), buf.get(1));
+    /// let mut vec_deque = VecDeque::new();
+    /// vec_deque.push_back('a');
+    /// vec_deque.push_back('b');
+    /// vec_deque.push_back('c');
+    ///
+    /// vec_deque.insert(1, 'd');
+    ///
+    /// let vec = vec_deque.iter().collect::<Vec<_>>();
+    /// assert_eq!(vec, [&'a', &'d', &'b', &'c']);
```

Comment:
Ah, I see the problem. Can you try changing `vec_deque.iter().collect::<Vec<_>>();` to `vec_deque.into_iter().collect::<Vec<_>>();`? That should fix the compile error.